### ISSUE TITLE
new gulcalc allocrule for total peril loss = maximum subperil loss

### ISSUE THREAD

**Issue by johcarter** [OasisLMF/ktools#91]
Add a new gulcalc allocrule where the total loss is the maximum of the subperil losses.

details of back-allocation rule to follow with input from impact forecasting

**Comment by johcarter** [comment]
**Allocation rule**

Allocate combined loss to the highest contributing sub peril. If there are multiple highest contributing subperils, distribute evenly between them.

**Example 1:** 
Peril A = 0.4
Peril B = 0.3
Peril C = 0.2

Combined loss  = 0.4

Allocated loss
Peril A = 0.4
Peril B = 0
Peril C = 0

**Example 2:**
Peril A = 0.4
Peril B = 0.3
Peril C = 0.4

Combined loss  = 0.4

Allocated loss
Peril A = 0.2
Peril B = 0
Peril C = 0.2

**Example 3 (total exposure):**
Peril A = 1
Peril B = 1
Peril C = 1

Combined loss  = 1

Allocated loss
Peril A = 0.3333
Peril B = 0.3333
Peril C = 0.3333
